src/apps/frontend/components/button/index.tsx

```diff
@@ -10,6 +10,8 @@ interface ButtonProps {
   onClick?: (e: React.MouseEvent<HTMLButtonElement>) => void;
   type?: ButtonType;
   kind?: ButtonKind;
+  startIcon?: React.ReactElement | string;
```

Comment:
Since these two does not apply for a large % of button's, it makes a poor choice to add them as property.

If we look at library such as baseweb, you can see similar pattern as well [here](https://baseweb.design/components/button/)

If you do see a need for a button where there is start, and end icon somewhat few times, its ok to create a new button class using this class via composition pattern.